### DISCUSSION TITLE
Migration to universal GLFW instead of Win32 only

### DISCUSSION THREAD

**Discussion by arakis** [amerkoleci/Vortice.Vulkan#27] [General]
Regarding PR #24 and Issue #26.

Thx for your reply. Good to know that you're planning to use GLFW for the future. Is there anything i can help?

My first draft was to use https://github.com/opentk/opentk/tree/master/src/OpenTK.Windowing.GraphicsLibraryFramework only (it is only the raw wrapper, without any helpers). What do you mean you have plans to use GLFW in an different way? In my opinion, it makes no sense to write an own wrapper, reinventing the wheel makes no sense.

But while crafting my first working draft with GLFW, i found out, it would be more better to use the Window framework of OpenTK directly (by the way, there's an open [PR](https://github.com/opentk/opentk/pull/1334) for supporting a special case for Vulkan, i submitted some days ago).

With OpenTK, you use existing structures (Vector2, Box2) and a you can (optionally) use the very powerfull Math Library.

Another consideration would be, just to use the new #24, as long you did not ported your own GLFW way. Because your current sample is windows only and is fully bloaded of platform specific boilter plate code. So, you will nothing "lost", if you would use my implenentation for the "meantime".

**Comment by amerkoleci** [comment]
Here we go:
https://github.com/amerkoleci/Vortice.Vulkan/commit/f97c6d43102e7b9b1cc2984c61fe5a3eb59ccb8c

I don't care that much about entire OpenTK and mathematics, just used the part I needed.

Thanks for bringing attention on this.
